### PR TITLE
[BE-Feat] 논의 참여(초대 수락) API 구현

### PR DESCRIPTION
**Pull request by efdao**
## #️⃣ 연관된 이슈>
- #174 

## 📝 작업 내용> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)
- password 인증 실패 횟수를 redis에 저장하고 사용합니다.
- joinDiscussion 메서드 구현하였습니다.
- DiscussionController에 초대 수락 api 매핑했습니다.
- 관련 테스트 코드 추가했습니다.

## 🙏 여기는 꼭 봐주세요! > 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
5번 실패시 403 forbidden으로 처리했습니다. 검토 부탁드려요

## Summary by CodeRabbit

- **New Features**
	- Introduced a new discussion join option that enables users to participate in discussions using password verification.
	- Updated password requirements to enforce a valid 4-6 digit numeric password.
	- Enhanced error handling to notify users after excessive failed attempts, improving the overall join experience.
	- Added new data transfer objects for join discussion requests and responses.

- **Bug Fixes**
	- Improved handling of invalid discussion IDs and null passwords, ensuring appropriate error messages are returned.

- **Tests**
	- Added comprehensive unit tests for discussion joining functionality and password handling scenarios, including edge cases for failed attempts.

- **Documentation**
	- Updated error codes to include specific messages for failed attempts and password requirements.

### CODE REVIEW

File: backend/src/main/java/endolphin/backend/global/redis/PasswordCountService.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        String redisKey = "failedAttempts:" + discussionId + ":" + userId;
+
+        String countStr = redisStringTemplate.opsForValue().get(redisKey);
+        int failedAttemptsCount = countStr != null ? Integer.parseInt(countStr) : 0;
+
+        if (failedAttemptsCount >= MAX_FAILED_ATTEMPTS) {
+            throw new ApiException(ErrorCode.TOO_MANY_FAILED_ATTEMPTS);
+        }
+
+        Long updatedCount = redisStringTemplate.opsForValue().increment(redisKey);
```
Comment (kwon204) [inline backend/src/main/java/endolphin/backend/global/redis/PasswordCountService.java:29]: just ask: 처음 시도한 사용자면 updatedCount에 null이 들어가나요?

Comment (efdao) [reply backend/src/main/java/endolphin/backend/global/redis/PasswordCountService.java:29]: 키에 해당하는 데이터가 존재하지 않으면 1이 들어갑니다

Comment (kwon204) [reply backend/src/main/java/endolphin/backend/global/redis/PasswordCountService.java:29]: 아 그렇군요

**Review by kwon204 — APPROVED** [review]
고생하셨습니다.
